benchmark/lib/store.js

```
@@ -1,39 +1,142 @@
     } else {
-      throw e;
+      logger.warn(`Error registering commit: ${error}`);
     }
   }
+}
+
+const persistAllScores = async () => {
+  const scoresTable = process.env.BENCH_SCORES_TABLE;
+
+  const now = new Date();
+  const timestamp = now.toISOString();
+  const date = timestamp.split('T')[0];
+
+  const commit = process.env.COMMIT || 'local';
+
+  if (commit == 'local') {
+    logger.debug(`Skipping putting aggregate scores in DynamoDB, logging to debug ${allScores.length} rows instead`);
+    logger.debug(updated)
+    return;
+  }
 
+  await registerCommit();
+
+  logger.debug(`Putting aggregate scores in DynamoDB with ${allScores.length} rows`);
+  const docClient = getDocClient();
+
+  // batch update scores using AWS DynamoDB DocumentClient
+  const BATCH_SIZE = 25; // DynamoDB batchWrite has a max of 25 items per batch
+
+  for (let i = 0; i < allScores.length; i += BATCH_SIZE) {
+    const batchItems = allScores.slice(i, i + BATCH_SIZE);
+    const requestItems = {
+      [scoresTable]: batchItems.map(score => ({
+        PutRequest: {
+          Item: score
+        }
+      }))
+    };
+
+    let retries = 2;
+    let unprocessedItems = requestItems;
+    let done = false;
+    while (!done) {
+      try {
+        const result = await docClient.batchWrite({ 
+          RequestItems: unprocessedItems 
+        });
+        if (result.UnprocessedItems && Object.keys(result.UnprocessedItems).length) {
+          logger.warn('Some items were not processed:', result.UnprocessedItems);
+          retries--;
+          unprocessedItems = result.UnprocessedItems
+        } else {
+          logger.debug('Batch write successful for scores:', batchItems.length);
+          done = true;
+        }
+      } catch (error) {
+        logger.warn('Batch write error:', error);
+        retries--;
+      }
+      if (retries <= 0) {
+        logger.error('Failed to write all score items to DB after retries');
+        done = true;
```

is this rare enough that we can just throw an error?